File: problems/2506_count_pairs_of_similar_strings/solution.cpp
#include <vector>
#include <string>
#include <unordered_map>
#include <unordered_set>
#include <algorithm>
#include <climits>
#include <queue>
#include <stack>
#include <cmath>
#include <numeric>
#include <set>
#include <map>
using namespace std;

class Solution {
public:
    int similarPairs(vector<string>& words) {
        int count = 0;
        auto mask = [](string& s) { int m = 0; for (char c : s) m |= 1 << (c - 'a'); return m; };
        for (int i = 0; i < words.size(); i++)
            for (int j = i + 1; j < words.size(); j++)
                if (mask(words[i]) == mask(words[j])) count++;
        return count;
    }
};

File: problems/2506_count_pairs_of_similar_strings/Solution.java
class Solution {
    public int similarPairs(String[] words) {
        int count = 0;
        for (int i = 0; i < words.length; i++) {
            int a = mask(words[i]);
            for (int j = i + 1; j < words.length; j++)
                if (a == mask(words[j])) count++;
        }
        return count;
    }
    int mask(String s) { int m = 0; for (char c : s.toCharArray()) m |= 1 << (c - 'a'); return m; }
}
